app/controllers/application_controller.rb

```diff
@@ -2,4 +2,9 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  def current_user
+    @_current_user ||= User.find session[:user_id] if session[:user_id]
```

Comment:
Would it be useful return `Guest.new` here if the `session[:user_id]` is nil? (My rule-of-thumb is that query methods should never return nil).